static/js/stream_list.js

```diff
@@ -335,6 +339,7 @@ class StreamSidebarRow {
 
     update_unread_count() {
         const count = unread.num_unread_for_stream(this.sub.stream_id);
+        // TODO: Do we need to check if there are unread mentions here?
         update_count_in_dom(this.$list_item, count);
     }
```

Comment:
Thanks @timabbott  for the updates on PR; I went through the changes made and did the analysis on the TODO part you mentioned;

There's no need to update the part `stream_has_any_unread_mention_messages` here as even on doing it we get the value as `false` always initially and then I think the function to calculate `counts` is run and we get the value updated as `true` by running `set_stream_unread_count`;

Reply:
I would also like to propose to remove this whole thing - `udpate_unread_count()` from `StreamSidebar` class as it always runs the function `update_count_in_dom` with `count=0` only and then again the actual count gets updated by `set_stream_unread_count`, so what we can do is ->

remove this complete function from `StreamSidebar` and add `:not(:empty)` to `.unread_count` class in CSS;

I tried running the application with these updated changes and it looked fine;
let me know if there's anything I missed which makes this proposal wrong;

Reply:
Yeah I think that's probably a fine approach, assuming you've confirmed that `count` is actually 0 -- how did you do that? I'd have expected it to depend on whether there are actually unreads if our initialization order is correct.

Reply:
I confirmed this by adding console statements; ->
Here's a screenshot which prints in order - `count`, `stream_id`, `place where console is written`;

<img width="480" alt="image" src="https://user-images.githubusercontent.com/76561593/185809325-1c4734d7-c79a-4256-9011-1aa72b953e80.png">


as you can see clearly that value of `count` is always 0 inside `update_unread_count()` and it gets updated inside `set_stream_unread_count()` which is called after this;